internal/services/containerapps/helpers/container_apps.go

```diff
@@ -2240,6 +2257,20 @@ func SecretsDataSourceSchema() *pluginsdk.Schema {
 		Sensitive: true,
 		Elem: &pluginsdk.Resource{
 			Schema: map[string]*pluginsdk.Schema{
+				"identity": {
+					Type:        pluginsdk.TypeString,
+					Computed:    true,
+					Sensitive:   false,
+					Description: "The identity to use for accessing key vault reference.",
```

Comment:
Can we add a validation to ensure it is either a user assigned identity id (via `commonids.ValidateUserAssignedIdentityID`), or a string that equals to `System` (for system managed identity)?

Reply:
Yep I've added this in now